Tensile/KernelWriter.py

```diff
@@ -556,7 +556,11 @@ def makeSubIterSchedule(self, kernel, localReadCode, iteration, pointerCode, wai
           if self.perIterLocalWriteCode[iteration-kernel["PrefetchLocalRead"]].countType(Code.LocalWriteInst) and kernel["PrefetchLocalRead"]:
             preIterLocalWrites = self.perIterLocalWriteCode[iteration-kernel["PrefetchLocalRead"]].countType(Code.LocalWriteInst)
             preIterGlobalReads = self.perIterGlobalReadCode[iteration-kernel["PrefetchLocalRead"]].countType(Code.GlobalReadInst)
-            skipPreIterLW = max(preIterLocalWrites - (numMfmaPerIter - preIterGlobalReads - 1),1) + max(numMfmaPerIter - readsPerIter, 0)
+            if preIterLocalWrites - (numMfmaPerIter - preIterGlobalReads - 1) < 0:
+               # condition for case in previous iteration writes scheduled < mfmaIter (we only got few slots for scheduling writes in previous writes)
+               skipPreIterLW = preIterLocalWrites - preIterGlobalReads
+            else:
+               skipPreIterLW = max(preIterLocalWrites - (numMfmaPerIter - preIterGlobalReads - 1),1) + max(numMfmaPerIter - readsPerIter, 0)
```

Comment:
can you test with (PR#969 https://github.com/ROCmSoftwarePlatform/Tensile/pull/969/commits/cf50db5bf40b0b7f3c839922e12a0ce32a31b0b8) to see if the bug still exist?

(since this commit has been squashed into PR972 and latest develop as well, It's not showing in PR969 before. I have made some change in PR969, use merge instead of rebase to develop, in order to show this commit)

Reply:
taken you pr # 969 into master. this fix is on top of that..